### DISCUSSION TITLE
; and , not working out of the box with flash.treesitter()

### DISCUSSION THREAD

**Discussion by cohml** [folke/flash.nvim#390] [General]
[The docs](https://github.com/folke/flash.nvim/blob/main/doc/flash.nvim.txt#L346C1-L348C78) state the following:

> - **Treesitter**`require("flash").treesitter(opts?)` opens **flash** in **Treesitter** mode
>   - use a jump label, or use `;` and `,` to increase/decrease the selection

To me, this suggests that upon simply calling `treesitter()` and selecting a node, `;` and `,` should increment/decrement the selection up/down the syntax tree.

But that's not what happens. Instead, `;` and `,` just jump to the last character searched with `f`/`t`, ignoring treesitter completely. This happens even when all other plugins besides `flash` are disabled.

I've tried multiple things to address this. But the docs suggest it should "just work", so I'm wary of reinventing the wheel.

Here is my config (`lazy` is my plugin manager):

```lua
local flash = require("flash")
flash.setup(
    {
        modes = {
            search = {
                enabled = true,
                highlight = {
                    backdrop = true,
                },
            },
        },
    }
)
vim.keymap.set(
    { "n", "x", "o" },
    "<leader>sf",
    function flash.treesitter() end,
    { desc = "Select with [F]lash", silent = true }
)
```

Any obvious issues?

**Comment by frnsys** [comment]
Did you ever figure this out? I can't get it working either.

**Reply by cohml** [reply]
Nope :(

**Comment by tfriedel** [comment]
This was tripping me up as well. I think we misunderstand how it's supposed to work:
> use a jump label, **or** use ; and , to increase/decrease the selection

I was pressing a jump label and **then** started pressing ; and ,  which didn't expand or shrink the selection. If I just press those keys **instead** of a jump label the selection expands or shrinks from my current position.

**Reply by cohml** [reply]
AHA! That's totally it!! The `or` is doing some heavy lifting there that I hadn't even realized.

This shows that the issue is not functionality, but documentation. @folke, I believe this awesome functionality vis-a-vis `,` / `;` could be more clearly explained in the docs. IMHO the current phrasing is ambiguous.

Edit: I think the discussion can be closed now.
